go/ql/lib/semmle/go/security/TaintedPathCustomizations.qll

```diff
@@ -91,19 +91,46 @@ module TaintedPath {
   }
 
   /**
-   * A call to `[file]path.Clean("/" + e)`, considered to sanitize `e` against path traversal.
+   * A call to `filepath.Clean("/" + e)`, considered to sanitize `e` against path traversal.
    */
   class FilepathCleanSanitizer extends Sanitizer {
     FilepathCleanSanitizer() {
       exists(DataFlow::CallNode cleanCall, StringOps::Concatenation concatNode |
         cleanCall =
-          any(Function f | f.hasQualifiedName(["path", "path/filepath"], "Clean")).getACall() and
+          any(Function f | f.hasQualifiedName("path/filepath", "Clean")).getACall() and
         concatNode = cleanCall.getArgument(0) and
         concatNode.getOperand(0).asExpr().(StringLit).getValue() = "/" and
         this = cleanCall.getResult()
       )
     }
   }
+      /**
+   * A call to `filepath.Base(e)`, considered to sanitize `e` against path traversal.
+   */
+  class FilepathBaseSanitizer extends Sanitizer {
+    FilepathBaseSanitizer() {
+      exists(Function f, FunctionOutput outp |
+        f.hasQualifiedName("path/filepath", "Base") and
+        outp.isResult(0) and
+        this = outp.getNode(f.getACall())
+      )
+    }
+  }
+
+  /** An call to ParseMultipartForm creates multipart.Form and cleans mutlpart.Form.FileHeader.Filename using path.Base() */
+  class MultipartClean extends Sanitizer {
+    MultipartClean() {
+      exists(DataFlow::FieldReadNode frn,  ControlFlow::Node node, DataFlow::CallNode cleanCall, Method get |
+        get.hasQualifiedName("net/http","Request", "ParseMultipartForm") and
+        cleanCall = get.getACall() and
+        cleanCall.asInstruction() = node and
+        frn.getField().hasQualifiedName("mime/multipart", "FileHeader", "Filename") and
+        node.getASuccessor*() = frn.asInstruction()
```

Comment:
Use `DataFlow::localFlow` to establish dataflow from the multipart parse to the access of the Filename field.

Also why are we sanitizing `frn.getBase()` (the `Form` that was read from), not `frn` itself (the `Filename` we read?)

Reply:
Fixed, also don't need flow here since according to documentation the field is only available after ParseMultipartForm is called.
```go
// MultipartForm is the parsed multipart form, including file uploads.
// This field is only available after ParseMultipartForm is called.
// The HTTP client ignores MultipartForm and uses Body instead.
MultipartForm *[multipart](https://pkg.go.dev/mime/multipart).[Form](https://pkg.go.dev/mime/multipart#Form)
```